SU2_CFD/include/numerics/turbulent/transition/trans_correlations.hpp

```diff
@@ -189,4 +190,107 @@ class TransLMCorrelations {
+
+    su2double rethetac = 0.0;
+
+    switch (options.Correlation_SLM) {
+      case TURB_TRANS_CORRELATION_SLM::MENTER_SLM: {
+
+        /*-- Thwaites parameter ---*/
+        // su2double  lambda_theta_local = 7.57e-3 * du_ds * wall_dist * wall_dist * Density / Laminar_Viscosity + 0.0128;
+        su2double  lambda_theta_local = lambda_theta;
+        lambda_theta_local = min(max(lambda_theta_local, -1.0), 1.0);
+
+        /*-- Function to sensitize the transition onset to the streamwise pressure gradient ---*/
+        su2double FPG = 0.0;
+        const su2double C_PG1 = 14.68;
+        const su2double C_PG1_lim = 1.5;
+        const su2double C_PG2 = -7.34;
+        const su2double C_PG2_lim = 3.0;
+        const su2double C_PG3 = 0.0;
+        if (lambda_theta_local >= 0.0) {
+          FPG = min(1+ C_PG1 * lambda_theta_local, C_PG1_lim);
+        } else {
+          const su2double FirstTerm = C_PG2 * lambda_theta_local;
+          const su2double SecondTerm = C_PG3 * min(lambda_theta_local + 0.0681, 0.0);
+          FPG = min(1 + FirstTerm + SecondTerm, C_PG2_lim);
+        }
+
+        FPG = max(FPG, 0.0);
+
+        const su2double C_TU1 = 100.0;
+        const su2double C_TU2 = 1000.0;
+        const su2double C_TU3 = 1.0;
+        rethetac = C_TU1 + C_TU2 * exp(-C_TU3 * Tu_L * FPG);
+
+        break;
+      } case TURB_TRANS_CORRELATION_SLM::CODER_SLM: {
+        
+        /*-- Local pressure gradient parameter ---*/
+        const su2double H_c = max(min(wall_dist * VorticityMag / VelocityMag, 1.1542), 0.3823);
+
+        /*-- Thwaites parameter ---*/
+        su2double lambda_theta_local = 0.0;
+        const su2double H_c_delta = 0.587743 - H_c;
+        if ( H_c >= 0.587743 ) {
+          const su2double FirstTerm = 0.1919 * pow(H_c_delta, 3.0);
+          const su2double SecondTerm = 0.4182 * pow(H_c_delta, 2.0);
+          const su2double ThirdTerm = 0.2959 * H_c_delta;
+          lambda_theta_local = FirstTerm + SecondTerm + ThirdTerm;
+        } else {
+          const su2double FirstTerm = 4.7596 * pow(H_c_delta, 3.0);
+          const su2double SecondTerm = -0.3837 * pow(H_c_delta, 2.0);
+          const su2double ThirdTerm = 0.3575 * H_c_delta;
+          lambda_theta_local = FirstTerm + SecondTerm + ThirdTerm;
+        }
+
+        /*-- Function to sensitize the transition onset to the streamwise pressure gradient ---*/
+        su2double FPG = 0.0;
+        if (lambda_theta_local <= 0.0) {
+          const su2double FirstTerm = -12.986 * lambda_theta_local;
+          const su2double SecondTerm = -123.66 * pow(lambda_theta_local, 2.0);
+          const su2double ThirdTerm = -405.689 * pow(lambda_theta_local, 3.0);
+          FPG = 1 - (FirstTerm + SecondTerm + ThirdTerm) * exp(-pow(Tu_L/1.5,1.5));
+        } else {
+          FPG = 1 + 0.275 * (1 - exp(-35.0 * lambda_theta_local)) * exp(-Tu_L/0.5);
+        }
+
+        // This is not reported in the paper
+        //FPG = max(FPG, 0.0);
+
```

Comment:
```suggestion
```

Reply:
I have left it there since I do not know if I have to keep it or not.